staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1463,6 +1470,13 @@ func mergeSlice(original, patch []interface{}, schema LookupPatchMeta, mergeKey
 			return nil, fmt.Errorf("cannot merge lists without merge key for %s", schema.Name())
 		}
 
+		if err = setMergeKeyForEmptyMapInSlice(original, mergeKey); err != nil {
+			return nil, err
+		}
+		if err = setMergeKeyForEmptyMapInSlice(patch, mergeKey); err != nil {
+			return nil, err
+		}
```

Comment:
similar questions about whether the assumptions in setMergeKeyForEmptyMapInSlice are correct for this call site (are items in the list always non-nil maps for which `mergeKey: nil` is a valid value to insert? the `deduplicateScalars` call above makes it seem like the lists could contain scalars, not just maps

Reply:
I think the condition `kind != reflect.Map` is to check if the item is map or not, so it's safe here. And the following calls (`mergeSliceWithSpecialElements` or `getMapAndMergeKeyValueByIndex` ) also have the map type assumption to item